### ISSUE TITLE
Is a standard library available?

### ISSUE THREAD

**Issue by njlr** [wasdk/WasmFiddle#4]
[This fiddle](https://wasdk.github.io/WasmFiddle//?12tngp) does not compile. 

It seems that some standard library files _are_ available, but it is not configured correctly. 

Which compiler is behind the fiddle? 
Which standard library features are available?

**Comment by mbebenita** [comment]
This is because WasmFiddle defaults to C, not C++. Use the wrench button to select C+11 and it should compile. However, you'll run into another problem, some symbols are not linked in. I believe we don't link in stdlib, is that correct @yurydelendik ?

**Comment by njlr** [comment]
Yes, switching to C++ [allows it to compile](https://wasdk.github.io/WasmFiddle//?sgpjd). However, I then get a link error: 

```
line 2: LinkError: import object field '_ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE6__initEPKcj' is not a Function
```

I guess the standard library would also need to be compiled to WebAssembly. Has this been done before? How feasible is this?

**Comment by mbebenita** [comment]
It has been done before. This is what Emscripten does for you, it compiles parts of the standard library and provides a sys-call emulation layer. One of the goals of the WasmFiddle was to allow developers to write short snippets of C/C++ code that compile to the smallest possible WebAssembly files, and that don't rely on importing large parts of the standard library. However, I do realize this has limited use so we may need to increase the scope here.

**Comment by yurydelendik** [comment]
There is an effort to have all standard functions in one "core" library -- see https://github.com/wasdk/wasmbase . But it's far from being production and it's not compatible with what llvm produces (vs emscripten).

**Comment by yurydelendik** [comment]
Standard library available with new compiler (see 59eb5782d0ba8e5439075b8c64ef3865480a906f). Closing as fixed. All development of WasmFiddle is moving to https://github.com/wasdk/WebAssemblyStudio .
